File: nodes/text-scratch-pad/text-scratch-pad.process.js
export default async ({inputs, settings, config, nodeConfig}) => {
  try {
    const text = inputs.text;
    const delimiter = inputs.delimiter || settings.default_delimiter || '\n';
    const clear = inputs.clear || false;
    
    // Get current accumulated content from settings
    let currentContent = settings.accumulated_content || settings.initial_content || '';
    
    // Clear if requested
    if (clear) {
      currentContent = '';
    }
    
    // Add new text if provided
    if (text !== undefined && text !== null && text !== '') {
      if (currentContent === '') {
        // First addition, no delimiter needed
        currentContent = String(text);
      } else {
        // Subsequent additions, use delimiter
        currentContent = currentContent + delimiter + String(text);
      }
    }
    
    // Update settings to persist the accumulated content
    const outputs = {
      content: currentContent,
      length: currentContent.length,
      __updated_settings: {
        accumulated_content: currentContent
      }
    };
    
    return outputs;
    
  } catch (error) {
    throw new Error(`Text Scratch Pad error: ${error.message}`);
  }
};

File: nodes/text-scratch-pad/text-scratch-pad.process.py
from typing import Any, Dict, List, Union

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    try:
        text = inputs.get('text')
        delimiter = inputs.get('delimiter') or settings.get('default_delimiter', '\n')
        clear = inputs.get('clear', False)
        
        # Get current accumulated content from settings
        current_content = settings.get('accumulated_content') or settings.get('initial_content', '')
        
        # Clear if requested
        if clear:
            current_content = ''
        
        # Add new text if provided
        if text is not None and text != '':
            if current_content == '':
                # First addition, no delimiter needed
                current_content = str(text)
            else:
                # Subsequent additions, use delimiter
                current_content = current_content + delimiter + str(text)
        
        # Update settings to persist the accumulated content
        outputs = {
            'content': current_content,
            'length': len(current_content),
            '__updated_settings': {
                'accumulated_content': current_content
            }
        }
        
        return outputs
        
    except Exception as error:
        raise Exception(f"Text Scratch Pad error: {str(error)}")
